cmd/termui/provider/metricsProvider.go

```diff
@@ -94,24 +89,24 @@ func (smp *StatusMetricsProvider) updateMetrics() {
 	smp.fetchAndApplyMetrics(statusMetricsUrlSuffix)
 	smp.fetchAndApplyBootstrapMetrics(bootstrapStatusMetricsUrlSuffix)
 
-	if smp.shardID != "" {
+	if smp.shardID != "" && smp.gatewayAddress != "" {
 		metricsURLSuffix := trieStatisticsMetricsUrlSuffix + smp.shardID
-		smp.fetchAndApplyGatewayStatusMetrics(metricsURLSuffix)
+		statusMetricsURL := smp.gatewayAddress + metricsURLSuffix
+		smp.fetchAndApplyGatewayStatusMetrics(statusMetricsURL)
```

Comment:
careful, this will be called each ~500 milliseconds. The `fetchAndApplyGatewayStatusMetrics` calls the `loadMetricsFromGatewayApi` function which does an API call, which is not free :)
Suggestion: I would call `loadMetricsFromGatewayApi` only if the gateway address changed (we might encounter this situation when the termui window was opened before the node, or there were some connection issues that were resolved). The result of this load would then be stored inside the presenter and used to actually output the trie sync string.

Reply:
added a separate bool check for this